inc/composer/class-command.php

```diff
@@ -654,14 +705,175 @@ protected function db( InputInterface $input, OutputInterface $output ) {
+
+				switch ( $os ) {
+					case 'Darwin':
+						$binary_arch = ( $arch === 'x86_64' ) ? 'darwin-amd64' : 'darwin-arm64';
+						break;
+					case 'Linux':
+						$binary_arch = ( $arch === 'amd64' ) ? 'linux-amd64' : 'linux-arm64';
+						break;
+					case self::is_wsl():
+						$binary_arch = 'windows-amd64.exe';
+						break;
+					default:
+						$binary_arch = null;
+						break;
+				}
+
+				// If couldn't detect a support architecture, ask the user to install mkcert manually.
+				if ( ! $binary_arch ) {
+					$output->writeln( '<error>This command is only supported on macOS, Linux, and Windows x64, install `mkcert` manually for other systems.</error>' );
+					$output->writeln( '<error>Download and setup `mkcert` from https://github.com/FiloSottile/mkcert </error>' );
+					return 1;
+				}
+
+				$binary = "mkcert-$mkcert_version-$binary_arch";
+				$mkcert = 'vendor/mkcert';
+
+				// Check if mkcert is installed globally already, bail if so.
+				$version = trim( shell_exec( 'mkcert -version' ) );
+				if ( $version ) {
+					$output->writeln( "<error>mkcert $version is installed globally already</error>" );
+					return 1;
+				}
+
+				// Check if mkcert is installed locally already, bail if so.
+				$version = trim( shell_exec( "$mkcert -version" ) );
+				if ( $version ) {
+					$output->writeln( "<error>mkcert $version is installed locally already</error>" );
+					return 1;
+				}
+
+				exec( "curl -o $mkcert -L https://github.com/FiloSottile/mkcert/releases/download/$mkcert_version/$binary", $dummy, $result );
+				if ( $result ) {
+					$output->writeln( '<error>Could not download mkcert binary, try using sudo or manually installing mkcert.</error>' );
+					$output->writeln( '<error>Download and setup `mkcert` from https://github.com/FiloSottile/mkcert </error>' );
+					return 1;
+				}
+
+				$output->writeln( "<info>mkcert $mkcert_version was downloaded.</info>" );
+
+				chmod( $mkcert, 0755 );
+
+				exec( "$mkcert -version", $dummy, $result );
+				if ( $result ) {
+					$output->writeln( '<error>Could not launch mkcert binary, try manually installing mkcert.</error>' );
+					$output->writeln( '<error>Download and setup `mkcert` from https://github.com/FiloSottile/mkcert </error>' );
+					return 1;
+				}
+				$output->writeln( "<info>mkcert $mkcert_version was installed.</info>" );
+
+				// Setup and accept the root certificate.
+				exec( "$mkcert -install", $dummy, $result );
+				if ( $result ) {
+					$output->writeln( '<error>Could not setup `mkcert` properly, try manually installing it.</error>' );
+					$output->writeln( '<error>Download and setup `mkcert` from https://github.com/FiloSottile/mkcert </error>' );
+					return 1;
+				}
+
+				$output->writeln( '<info>mkcert root CA was installed and accepted successfully.</info>' );
+				break;
+			case 'generate':
+				// TODO figure out how to programmatically detect the domains to use.
+				$domains = $input->getArgument( 'options' )[1] ?? '*.altis.dev';
```

Comment:
I think the default behaviour here should be the following:

- Always generate certs for `<project>.<tld>` and `*.<project>.<tld>` (no harm in them being there by default, even if someone calls the function with other domains, its better to avoid surprises here IMO than assume someone knows it might be a deleterious action if we don't explicitly call that out)
- Add support for a list of hosts in the config and merge those in by default too, we can document that custom ones must be pointed to localhost

Reply:
Agreed :+1:

Reply:
Traefik will need to know about the extra domains too 🤔

Reply:
Added the extra domains support to nginx ⬆️

Reply:
Also added the configured domain(s) to the domains within `generate()` command as discussed, also `altis.dev *.altis.dev`.